packages/dds/tree/src/simple-tree/api/snapshotCompatibilityChecker.ts

```diff
@@ -0,0 +1,112 @@
+ *
+ *	   // The current schema has a new optional field that was not present on Point2D
+ *	   z: factory.optional(factory.number),
+ * }) {}
+ * const oldViewSchema = importCompatibilitySchemaSnapshot(fs.readFileSync("Point2D.json"));
+ *
+ * // Check to see if the document created by the historical view schema can be opened with the current view schema
+ * const compatibilityStatus = checkCompatibility(oldViewSchema, currentViewSchema);
+ *
+ * // z is not present in Point2D, so the schema must be upgraded
+ * assert.equal(compatibilityStatus.canView, false);
+ *
+ * // The schema can be upgraded to add the new optional field
+ * assert.equal(compatibilityStatus.canUpgrade, true);
+ * ```
+ *
+ * @param viewWhichCreatedDocument - The view schema which was used to create the document.
+ * @param view - The view schema being used to read the document.
+ * @returns The compatibility status.
+ *
+ * @alpha
+ */
+export function checkCompatibility(
+	viewWhichCreatedDocument: TreeViewConfiguration,
+	view: TreeViewConfiguration,
+): Omit<SchemaCompatibilityStatus, "canInitialize"> {
+	const viewAsAlpha = new TreeViewConfigurationAlpha({ schema: view.schema });
+	const stored = toStoredSchema(viewWhichCreatedDocument.schema, {
+		includeStaged: () => true,
+	});
+	const tester = new SchemaCompatibilityTester(viewAsAlpha);
+	return tester.checkCompatibility(stored);
+}
+
+/**
+ * Returns a JSON representation of the tree schema for snapshot compatibility checking.
+ * @param config - The schema to snapshot. Only the schema field of the `TreeViewConfiguration` is used.
+ * @returns The JSON representation of the schema.
+ *
+ * @example This example creates and persists a snapshot of a Point2D schema.
+ *
+ * ```ts
+ * const schemaFactory = new SchemaFactory("test");
+ * class Point2D extends schemaFactory.object("Point", {
+ *     x: factory.number,
+ *     y: factory.number,
+ * }) {}
+ * const viewSchema = new TreeViewConfiguration({ schema: Point2D });
+ * fs.writeFileSync("Point2D.json", exportCompatibilitySchemaSnapshot(viewSchema));
+ * ```
+ *
+ * @alpha
+ */
+export function exportCompatibilitySchemaSnapshot(
+	config: Pick<TreeViewConfiguration, "schema">,
+): JsonCompatibleReadOnly {
+	const simpleSchema = toSimpleTreeSchema(config.schema, true);
+	return encodeSimpleSchema(simpleSchema);
+}
+
+/**
+ * Parse a JSON representation of a tree schema into a concrete schema.
+ * @param config - The JSON representation of the schema.
+ * @returns The schema. Only the schema field of the {@link TreeViewConfiguration} is populated.
+ * @throws Will throw a usage error if the encoded schema is not in the expected format.
+ *
+ * @example This example loads and parses a snapshot of a Point2D schema.
+ *
+ * ```ts
+ * const oldViewSchema = importCompatibilitySchemaSnapshot(fs.readFileSync("Point2D.json"));
+ * ```
+ * @privateRemarks We construct a TreeViewConfiguration here with the default parameters. The default set of validation parameters are
```

Comment:
remarks about code inside of a function (not about its signature) should be in comments in the implementation, not in private remarks for the function.